src/os/bluestore/StupidAllocator.cc

```diff
@@ -98,6 +98,7 @@ int StupidAllocator::allocate(
   int orig_bin = bin;
 
   auto p = free[0].begin();
+  auto end = free[0].end();
```

Comment:
better off moving the definition of `end` to where it is used. BTW, its value is always overwritten there.